pennylane/qnode.py

```diff
@@ -307,6 +350,11 @@ def construct(self, args, kwargs=None):
         # whether they are qubit or CV devices, and remove this logic here.
         self.type = 'CV' if all(are_cvs) else 'qubit'
 
+        if self.device.operations:
+            # replace operations in the queue with any decompositions if required
+            self.queue = decompose_queue(self.queue, self.device)
+            self.ops = self.queue + self.ev
```

Comment:
Might as well remove the similar line from 337 and de-indent this line out of conditional

Reply:
:+1:

Reply:
Done